perfmetrics/scripts/ls_metrics/listing_benchmark.py

```diff
@@ -0,0 +1,473 @@
+"""
+
+import argparse
+import ast
+import configparser
+import json
+import logging
+import os
+import statistics as stat
+import subprocess
+import sys
+import time
+
+import directory_pb2 as directory_proto
+
+sys.path.insert(0, '..')
+import generate_files
+import numpy as np
+import texttable
+
+from google.protobuf.json_format import Parse, ParseDict
+
+
+logging.basicConfig(
+    level=logging.INFO,
+    format='%(asctime)s [%(levelname)s] %(message)s',
+    handlers=[logging.StreamHandler(sys.stdout)],
+)
+log = logging.getLogger()
+
+
+def OutputResults(folders, results_list, message, num_samples) -> dict:
+  """Outputs the results on the console.
+
+  This function takes in dictionary containing the list of results (for all samples) for each
+  testing folder, for both the gcs bucket and persistent disk. Then it generates various metrics
+  out of these lists and outputs them into the console.
+
+  The metrics present in the output are (in msec):
+  Mean, Median, Standard Dev, 0th %ile, 20th %ile, 40th %ile, 60th %ile, 80th %ile, 90th %ile,
+  95th %ile, 98th %ile, 99th %ile, 100th %ile.
+
+  Args:
+    folders: List containing protobufs of testing folders.
+    results_list: Dictionary containing the list of results (for all samples)
+                  for each testing folder.
+    message: String which describes/titles the test.
+    num_samples: Number of samples to collect for each test.
+
+  Returns:
+    A dictionary containing the various metrics in a JSON format.
+  """
+  metrics = dict()
+
+  for testing_folder in folders:
+    metrics[testing_folder.name] = dict()
+
+    metrics[testing_folder.name]['Test Desc.'] = message
+    metrics[testing_folder.name]['Number of samples'] = num_samples
+    results_list[testing_folder.name] = sorted(results_list[testing_folder.name])
+
+    metrics[testing_folder.name]['Mean'] = stat.mean(results_list[testing_folder.name])
+    metrics[testing_folder.name]['Median'] = stat.median(results_list[testing_folder.name])
+    metrics[testing_folder.name]['Standard Dev'] = stat.stdev(gcs_bucket_results[testing_folder.name])
+
+    metrics[testing_folder.name]['Quantiles'] = dict()
+    for percentile in range(0, 100, 20):
+      metrics[testing_folder.name]['Quantiles']['{} %ile'.format(percentile)] = np.percentile(results_list[testing_folder.name], percentile)
+
+    metrics[testing_folder.name]['Quantiles']['90 %ile'] = np.percentile(results_list[testing_folder.name], 90)
+    metrics[testing_folder.name]['Quantiles']['95 %ile'] = np.percentile(results_list[testing_folder.name], 95)
+    metrics[testing_folder.name]['Quantiles']['98 %ile'] = np.percentile(results_list[testing_folder.name], 98)
+    metrics[testing_folder.name]['Quantiles']['99 %ile'] = np.percentile(results_list[testing_folder.name], 99)
+    metrics[testing_folder.name]['Quantiles']['100 %ile'] = np.percentile(results_list[testing_folder.name], 100)
+
+  print(metrics)
```

Comment:
is this just for logging?

Reply:
yes

Reply:
what is the difference between print and logger? why are you not using logger here